jupyterhub/values.yaml

```diff
@@ -480,7 +481,6 @@ prePuller:
       allowPrivilegeEscalation: false
     podSchedulingWaitDuration: 10
     nodeSelector: {}
-    tolerations: []
```

Comment:
Replaced prePuller.hook.tolerations for prePuller.extraTolerations because both daemonsets/jobs should have similar/same tolerations

Reply:
It is not obvious to the location of configuration etc, but prePuller.hook.tolerations is currently just for the hook-image-awaiter, which is a single pod that monitors the readiness state of the hook-image-puller daemonset's pods.

I think both deamonsets (hook-image-puller, continuous-image-puller) should have the same tolerations, but the hook-image-awaiter Job should have its own still.

Reply:
@consideRatio  Are you proposing to keep prePuller.hook.tolerations, and add prePuller.extraTolerations tolerations for deamonsets (hook-image-puller, continuous-image-puller) and also the single pod hook-image-awaiter?

Reply:
> Are you proposing to keep prePuller.hook.tolerations, and add prePuller.extraTolerations tolerations for deamonsets (hook-image-puller, continuous-image-puller) and also the single pod hook-image-awaiter?

Yes, but I don't think it makes sense to add the tolerations also for the image-awaiter Job's pod. I find the image-awaiter to be something that make sense to run on nodes like the hub pod etc rather than end up on a node that is busy pulling images and potentially blocked by image pulling in the worst case.

Reply:
what about only having then prePuller.continuous.tolerations?

Reply:
I'm open to relocating image-awaiter job's configuration someplace so its less confusing, because it sure is! For example as part of prePuller.hook.imageAwaiter.tolerations. This kind of change would need to be accompanied with a deprecation plan as well though, and we don't have very clear procedures for that at the moment, but typically by adding a fail statement in NOTES.txt or the more complicated option of adding a warning in NOTES.txt followed by logic to migrate the old configuration to work with the new config.

But without a strong case for having dedicated tolerations for only the continuous/hook image-puller daemonsets' pods, I don't want to set tolerations on the daemonset-specific level.

If one goes for DS-specific config, we end up maybe wanting to have 3 config options instead of one. DS-unspecific, and DS-specific x2, and that is a lot more work to ensure they merge with eachother in a sensible way etc and document and communicate that - it is simply more complicated without clear benefits I think.